doc/dev/adapter-development-walkthrough.md

```diff
@@ -0,0 +1,807 @@
+
+```
+package mysampleadapter
+
+import (
+	"io"
+	"log"
+	"testing"
+
+	"golang.org/x/net/context"
+
+	mixerapi "istio.io/api/mixer/v1"
+	"istio.io/mixer/pkg/adapter"
+	"istio.io/mixer/test/testenv"
+	"istio.io/mixer/template"
+	"path/filepath"
+)
+
+func TestMySampleAdapter(t *testing.T) {
+	operatorCnfg,err := filepath.Abs("sampleoperatorconfig")
+	if err != nil {
+		t.Fatalf("fail to get absolute path for sampleoperatorconfig: %v", err)
+	}
+
+	var args = testenv.Args{
+		// Start Mixer server on a free port on loop back interface
+		MixerServerAddr:               `127.0.0.1:0`,
+		ConfigStoreURL:                `fs://` + operatorCnfg,
+		ConfigStore2URL:               `fs://` + operatorCnfg,
+		ConfigDefaultNamespace:        "istio-config-default",
+		ConfigIdentityAttribute:       "destination.service",
+		ConfigIdentityAttributeDomain: "svc.cluster.local",
+	}
+
+	env, err := testenv.NewEnv(&args, template.SupportedTmplInfo, []adapter.InfoFn{GetInfo})
+	if err != nil {
+		t.Fatalf("fail to create testenv: %v", err)
+	}
+	defer closeHelper(env)
+
+	client, conn, err := env.CreateMixerClient()
+	if err != nil {
+		t.Fatalf("fail to create client connection: %v", err)
+	}
+	defer closeHelper(conn)
+
+	attrs := map[string]interface{}{"response.code": int64(400)}
+	bag := testenv.GetAttrBag(attrs, args.ConfigIdentityAttribute, args.ConfigIdentityAttributeDomain)
+	request := mixerapi.ReportRequest{Attributes: []mixerapi.Attributes{ bag}}
+	_, err = client.Report(context.Background(), &request)
+	if err != nil {
+		t.Errorf("fail to send report to Mixer %v", err)
+	}
+}
+
+func closeHelper(c io.Closer) {
+	err := c.Close()
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+```
+
+Add go_test to the BUILD file adapter/mysampleadapter/BUILD.
+
+Copy the following content into the existing BUILD file. The text in bold shows the newly added content.
+
+<pre>
+package(default_visibility = ["//visibility:public"])
+
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["mysampleadapter.go"],
+    visibility = ["//visibility:public"],
+    deps = [
+        "//adapter/mysampleadapter/config:go_default_library",
+        "//pkg/adapter:go_default_library",
+        "//template/metric:go_default_library",
+        "@com_github_gogo_protobuf//types:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@com_github_googleapis_googleapis//:google/rpc",
+    ],
+)
+
+<b>
+load("@io_bazel_rules_go//go:def.bzl", "go_test")
+go_test(
+    name = "go_default_test",
+    size = "small",
+    srcs = ["mysampleadapter_test.go"],
+    library = ":go_default_library",
+    data = [
+        "sampleoperatorconfig",
+    ],
+    deps = [
+        "//pkg/adapter:go_default_library",
+        "//pkg/template:go_default_library",
+        "//test/testenv:go_default_library",
+        "//template:go_default_library",
+        "@com_github_istio_api//:mixer/v1",  # keep
```

Comment:
please update to `@io_istio_api` as that would be the long-term name for this.

Reply:
Hi @kyessenov , this name is already specified in the WORKSPACE file of the Mixer, which this walkthrough task just assumes is there. I think when we do a find and replace on this name, this md file will also be updated as part of the global rename.

Reply:
The competing PR changes it (https://github.com/istio/mixer/pull/1367). Whichever lands first, we'll need to patch up the paths here.